pkg/v1/tarball/write.go

```diff
@@ -171,45 +223,26 @@ func MultiRefWrite(refToImage map[name.Reference]v1.Image, w io.Writer) error {
 			LayerSources: layerSources,
 		})
 	}
-
-	mBytes, err := json.Marshal(m)
-	if err != nil {
-		return err
-	}
-	return writeTarEntry(tf, "manifest.json", bytes.NewReader(mBytes), int64(len(mBytes)))
+	return m, nil
 }
 
-func dedupRefToImage(refToImage map[name.Reference]v1.Image) map[v1.Image][]string {
-	imageToTags := make(map[v1.Image][]string)
+// MultiRefTarManifest get the manifest.json that will be written to the tarball
+// for multiple references
+func MultiRefTarManifest(refToImage map[name.Reference]v1.Image) (Manifest, error) {
+	return processImages(refToImage, nil)
+}
 
-	for ref, img := range refToImage {
-		if tag, ok := ref.(name.Tag); ok {
-			if tags, ok := imageToTags[img]; ok && tags != nil {
-				imageToTags[img] = append(tags, tag.String())
-			} else {
-				imageToTags[img] = []string{tag.String()}
-			}
-		} else {
-			if _, ok := imageToTags[img]; !ok {
-				imageToTags[img] = nil
-			}
-		}
+// MultiTarManifest get the manifest.json that will be written to the tarball
+// for multiple tags
+func MultiTarManifest(tagToImage map[name.Tag]v1.Image) (Manifest, error) {
+	refToImage := make(map[name.Reference]v1.Image, len(tagToImage))
+	for i, d := range tagToImage {
+		refToImage[i] = d
 	}
-
-	return imageToTags
+	return MultiRefTarManifest(refToImage)
 }
 
-// Writes a file to the provided writer with a corresponding tar header
-func writeTarEntry(tf *tar.Writer, path string, r io.Reader, size int64) error {
-	hdr := &tar.Header{
-		Mode:     0644,
-		Typeflag: tar.TypeReg,
-		Size:     size,
-		Name:     path,
-	}
-	if err := tf.WriteHeader(hdr); err != nil {
-		return err
-	}
-	_, err := io.Copy(tf, r)
-	return err
+// TarManifest is a wrapper to get the Manifest for a single image and tag
+func TarManifest(ref name.Reference, img v1.Image) (Manifest, error) {
```

Comment:
I'd like to avoid adding too many of these entrypoints to avoid having to maintain them. I'd rename this to `ComputeManifest` (to avoid the stutter with `tarball.TarManifest`) and change it to use the signature from `MultiRefTarManifest` (which I think is generic enough to cover all use cases).

Reply:
I have no argument with this; actually, I think it makes sense to rename them, as they really are just computing the manifest.

I am going to do a bit of rejiggering on the basis of this. Expect it shortly.

Reply:
Changed the func names to match but haven't yet reduced the entrypoints. I put those in to parallel the `Write`/`MultiWrite`/`MultiRefWrite` and `WriteToFile`/`MultiWriteToFile`/`MultiRefWriteToFile`. I am fine if you want just `ComputeManifest`, but it will look a bit off given that the arguments are similar.

Totally your call; let me know.

Reply:
Yeah I think we should just reduce this down to one. The MultiWrite/MultiRefWrite thing is a historical mistake (I don't think we should ever have had the tag version), and calling ComputeManifest should be a niche enough thing that I don't think we need any sugar on top of it -- it's not too hard for callers to just wrap an `Image` in a map.

Reply:
Sure. Consider it done.